### PR TITLE
API - Add Get module list and get module field list endpoints

### LINKED ISSUE #7174: /Api/V8 needs the ability to return a list of modules
#### Issue

In `v4` and `/api/v8` you had the ability to get a list of all available modules.  Currently this functionality is missing from `/Api/V8`. It is a major blocker for plugin integrations which need to provide the user with a choice of modules they can connect to.

**Note** I'm splitting this issue off from #7083

#### Expected Behavior

Call `/Api/V8/modules` (or something else) and receive a list of available modules.

#### Actual Behavior

This functionality is missing.

#### Possible Fix

`/api/v8/` used the path `/modules/meta/list`

#### Context

Unable to list modules for integrations

#### Your Environment

* SuiteCRM Version used: 7.11.3

### LINKED ISSUE #7175: /Api/V8 needs the ability to a list of module's fields
#### Issue

In `v4` and `/api/v8` you had the ability to get a list of all available fields for a given module.  Currently this functionality is missing from `/Api/V8`. It is a major blocker for plugin integrations which need to provide the user with a choice of fields in a module they can connect to.

**Note** I'm splitting this issue off from #7083

#### Expected Behavior

Call `/Api/V8/modules/Accounts/fields` (or something else) and receive a list of available modules.

#### Actual Behavior

This functionality is missing.

#### Possible Fix

`/api/v8/` used the path `/modules/Accounts/meta/language`

#### Context

Unable to list fields for a module for integrations

#### Your Environment

* SuiteCRM Version used: 7.11.3

### PR DESCRIPTION
**Pull request by PedroErnst**
Adds two new endpoints to the API.

## Description

`V8/meta/modules`
Returns a list of modules

`V8/meta/fields/{moduleName}`
Returns a list of fields for this module, with field types

## Motivation and Context

Fix #7174 
Fix #7175 

## How To Test This

1. GET `V8/meta/modules` and see it returns a complete list of modules
2. GET `V8/meta/fields/Accounts` and see it returns all the fields for this module 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://docs.suitecrm.com/community/contributing-code/coding-standards/).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://docs.suitecrm.com/community/contributing-code/) guidelines.

### CODE REVIEW

**Comment by cameronblaikie** [comment]
Assessed :+1:

**Comment by PedroErnst** [comment]
Good point @Jason-Dang

File: Api/V8/Controller/MetaController.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+use Api\V8\Param\GetFieldListParams;
+use Api\V8\Service\MetaService;
+use Api\V8\Service\UserService;
+use Exception;
+use Slim\Http\Request;
+use Slim\Http\Response;
+
+/**
+ * UserController
```
Comment (connorshea) [inline Api/V8/Controller/MetaController.php]: Code comment is incorrect.

File: Api/V8/Service/MetaService.php [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        $bean = $this->beanManager->newBeanSafe($module);
+        $fieldList = [];
+        foreach ($bean->field_defs as $fieldName => $fieldDef) {
+            $fieldList[$fieldName] = $this->pruneVardef($fieldDef);
+        }
+        return $fieldList;
+    }
+
+    /**
+     * @param array $def
```
Comment (connorshea) [inline Api/V8/Service/MetaService.php]: It'd be nice to have comments explaining what the functions actually do. :)

File: Api/V8/Helper/ModuleListProvider.php [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+{
+    /**
+     * @return array
+     */
+    public function getModuleList()
+    {
+        global $current_user;
+
+        $modules = query_module_access_list($current_user);
+        \ACLController:: filterModuleList($modules, false);
```
Comment (connorshea) [inline Api/V8/Helper/ModuleListProvider.php]: I don't think this space should be here?

```suggestion
        \ACLController::filterModuleList($modules, false);
```

**Comment by PedroErnst** [comment]
Thanks for the input @connorshea

**Comment by PedroErnst** [comment]
The modules endpoint now returns a list of ACL enabled actions.

Example:
image

**Comment by cameronblaikie** [comment]
Assessed :+1:

**Pull request merged**